src/axom/spin/internal/linear_bvh/math.hpp

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_SPIN_BVH_MATH_H_
+#define AXOM_SPIN_BVH_MATH_H_
+
+#include "axom/core/Macros.hpp"
+#include "axom/core/Types.hpp"    // for fixed bitwidth types
+
+// include math so we can use functions defined
+// in both cuda and c
+#include <math.h>
+
+
+#ifndef __CUDACC__
+// make sure min / max resolve for both cuda and cpu
+#include <math.h>
+#include <string.h> //resolve memcpy
+using namespace std;
+#endif
+
+namespace axom
+{
+namespace spin
+{
+namespace internal
+{
+namespace linear_bvh
+{
+
+constexpr axom::int32 AXOM_INF_32    = 0x7f800000U;
+constexpr axom::int32 AXOM_NG_INF_32 = 0xff800000U;
+
+constexpr axom::int64 AXOM_INF_64    = 0x7ff0000000000000ULL;
+constexpr axom::int64 AXOM_NG_INF_64 = 0xfff0000000000000ULL;
+
+constexpr axom::int32  AXOM_NAN_32   = 0x7FC00000U;
+constexpr axom::int64 AXOM_NAN_64    = 0x7FF8000000000000ULL;
+
+constexpr float  AXOM_EPSILON_32 = 1e-5f;
+constexpr double AXOM_EPSILON_64 = 1e-9f;
+
+union Bits32
```

Comment:
I'm not sure how these unions are being used, but they should be used with care because unions in c++ behave differently than unions in c: "it's undefined behavior to read from the member of the union that wasn't most recently written".
https://en.cppreference.com/w/cpp/language/union

Reply:
We will refactor/unify all this in the future -- open to suggestions.